app/models/lookup_keys/puppetclass_lookup_key.rb

```diff
@@ -20,6 +20,10 @@ class PuppetclassLookupKey < LookupKey
 
   scope :smart_class_parameters, -> { joins(:environment_classes).readonly(false) }
 
+  def editable_by_user?
+    PuppetclassLookupKey.authorized(:edit_external_parameters).present? && PuppetclassLookupKey.authorized(:edit_external_parameters).find(self.id)
```

Comment:
This will cause two queries on the DB, better to do something like:
```ruby
PuppetclassLookupKey.authorized(:edit_external_parameters).where(:id => id).exists?
```
which will generate a much simpler query.